harness/determined/common/experimental/determined.py

```diff
@@ -124,15 +124,10 @@ def logout(self) -> None:
 
         authentication.logout(self._session._master, user, self._session._cert)
 
-    def list_users(self) -> Sequence[user.User]:
+    def list_users(self) -> Iterable[user.User]:
```

Comment:
I don't remember where exactly we landed in the discussion we had earlier, and if I'm just rehashing an old argument then please ignore.

Like you, I _do_ like returning an iterator, but it's a little unexpected to me for a function named `list_foo` to return one. I expect an iterator-returning function of users to be called just plain `users`.

Since there's some ambiguity, if you end up keeping this function's name and return type, could you please change the annotation to either `Iterator` or `Generator` to help clarify expectations to a reader?